### ISSUE TITLE
initial setup question: telnet

### ISSUE THREAD

**Issue by neophyte-savant** [sparkfun/phant#137]
I have command prompt displaying the http and telnet server information, but i cannot get another window to accept the telnet (with and without $)( command. running directly in node js just results in 3 dots. 

windows 7 64-bit. 

thanks

**Comment by neophyte-savant** [comment]
I have no idea what i changed, but i now sort of have a connection to the telnet. However, the formatting is quite messed up (with the character elephant) and all character key strokes are automatically entered, which gives an unrecognized input error. IE: typing any key is registered as if i hit enter right after i hit the key.

**Comment by neophyte-savant** [comment]
So i see the package.json was updated to 6.8, but im getting the same error problem. Here is a screenshot of the install and telnet launch, and the telnet response

![image](https://cloud.githubusercontent.com/assets/10458867/5810646/cbcecf70-a017-11e4-9497-6246d4b0d0e7.png)

![image](https://cloud.githubusercontent.com/assets/10458867/5810638/b5c55708-a017-11e4-89b9-fc7d08f0c1e6.png)

if anyone has any ideas, and general help for how to port this over to a shared server someplace, i would greatly appreciate it. 

thanks

**Comment by toddtreece** [comment]
@neophyte-savant try loading http://localhost:8080 in a web browser. i enabled the http manager by default for you w/ 6.8 so it's easier to get going

**Comment by neophyte-savant** [comment]
I got a 404 error. I ran "npm uninstall -g phant" then redid the installation, same result. Using chrome.

**Comment by toddtreece** [comment]
@neophyte-savant i'm guessing you restarted the server after re-install, but try http://127.0.0.1:8080 and see if that works. i'm not very familiar with running node on windows, so i'm just guessing here. i'll try to get someone from sparkfun to help you debug this on a windows box

**Comment by neophyte-savant** [comment]
still getting a 404 unfortunately.

**Comment by neophyte-savant** [comment]
There is a very simple answer to all of this: simply install Ubuntu!! I loaded up Ubuntu desktop onto a thumb drive, hit my head against a wall for a few minutes trying to get nodejs installed and updated (finally found a good repository that i didnt write down... idiot!)... and then the server side functioned perfectly. I need to get my arduino booted up again - ive been in perf-board mode trying to make everything a little more permanent. when i get that finished (probably next week) i will report back to this issue thread. 

secondary question: once i have everything working here - can i migrate it to a shared server (eg. godaddy - who hosts the main website and is handling the domain name etc)? Would i be better off trying to host the server myself on a standard desktop computer? Even if this gets deployed there wont be a huge amount of traffic - barely megabytes a day. I would just need godaddy to point towards my own server - which would mean working through our business' ISP more than anything else, right?

**Comment by jbdatko** [comment]
re secondary question: You'll need a hosting provider that provides you a
shell and root access. I don't use GoDaddy, but iirc, they don't do this
except for VPS. Which, if you have, then you are fine. Otherwise, I'm quite
happy with Digital Ocean which is $5 a month.

You only need to do this if your Arduino needs to reach a public Internet
server. If your Arduino is behind a NAT device, on a private network, than
you can run a private Phant instance on a desktop/Pi/BeagleBone/whatever
and that will all be fine. You could make your phant server Internet
visible and that's going to require some NAT forwarding.

In short, yes you can migrate this to shared server. But the specifics of
how to set this up on your network will be very specific to your personal
situation and goals.

**Comment by neophyte-savant** [comment]
Wow thanks for the quick/late night response. At some point this will go live and i'll need internet connectivity. i'll cross that bridge when it comes i guess. thanks for the note on digital ocean - godaddy wants $30/mo.

**Comment by neophyte-savant** [comment]
Went for broke and tried to install on ocean digital, says no such file or directory when running "phant". i ran the nodejs install as per the ocean digital page here: https://www.digitalocean.com/community/tutorials/how-to-install-node-js-on-an-ubuntu-14-04-server
is this version of node out date? 

![image](https://cloud.githubusercontent.com/assets/10458867/5968858/7d663b4a-a7ea-11e4-8ac5-e8a4286670a7.png)
